source/Patches/Roles/Role.cs

```diff
@@ -15,23 +17,39 @@
 
 namespace TownOfUs.Roles
 {
-    public abstract class Role
+    public abstract class Role : IRoleDetails
     {
+        private static List<Role> _roleSingletons { get; } = new List<Role>();
```

Comment:
I am nervous that this lives here. Mutable objects make me nervous. Can this state be instantiated with all the classes and held there immutably?

Reply:
`_roleSingletons` are private and can be accessed as `ReadOnlyCollection<Role> Role.RoleSingleton`,. If you think `_roleSingletons` can be mistakenly access inside the class, perhaps we can make another static class that holds all role informations or use different kind of type?

Reply:
You intend that `AddSingleton` is only called from the one specific place at the one specific time. But nothing in the code enforces it. So somebody could come along six months ago and call it when they shouldn't. It would be cleaner if `RoleSingleton` didn't call into the `Role` class, but instead it created this thing and guaranteed no one else would modify it. It might make sense for the accessor methods to live there too; there's too much stuff going on in `Role.cs` as it is.